pkg/agent/agent_windows.go

```diff
@@ -218,6 +285,78 @@ func (i *Initializer) prepareOVSBridgeOnHNSNetwork() error {
 	return nil
 }
 
+func (i *Initializer) prepareOVSBridgeForVM() error {
+	klog.InfoS("Performing OVS configuration", "hostIFName", i.nodeConfig.UplinkNetConfig.Name)
+	hostIFName := i.nodeConfig.UplinkNetConfig.Name
+	uplinkIFName := util.GenerateUplinkInterfaceName(hostIFName)
+	ovsPorts, ovsErr := i.ovsBridgeClient.GetPortList()
+	if ovsErr != nil {
+		return fmt.Errorf("failed to list OVS ports: %v", ovsErr)
+	}
+	for _, port := range ovsPorts {
+		if port.Name == hostIFName {
+			klog.Info("Uplink and host interface configuration exist in OVS")
+			return nil
+		}
+	}
+
+	success := false
+	uplinkExternalIDs := map[string]interface{}{
+		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaUplink,
+	}
+	// TODO: Have a separate function for creation of pair ports
+	// Create uplink port on OVS.
+	uplinkUUID, ovsErr := i.ovsBridgeClient.CreatePort(uplinkIFName, uplinkIFName, uplinkExternalIDs)
+	if ovsErr != nil {
+		return fmt.Errorf("failed to create uplink port on OVS for %s: %v", uplinkIFName, ovsErr)
+	}
+
+	// Manual clean up of OVS configurations is required, when agent exits
+	// abruptly or when the auto cleanup operation fails.
+	defer func() {
+		if !success {
+			klog.InfoS("Deleting port on OVS", "uplinkUUID", uplinkUUID)
+			if ovsErr := i.ovsBridgeClient.DeletePort(uplinkUUID); ovsErr != nil {
+				klog.ErrorS(ovsErr, "Failed to delete port on OVS", "uplinkUUID", uplinkUUID)
+			}
+		}
+	}()
+
+	// Query the uplink port to check if its created
+	uplinkOFPort, ovsErr := i.ovsBridgeClient.GetOFPort(uplinkIFName, false)
+	if ovsErr != nil {
+		return fmt.Errorf("failed to create uplink ofport on OVS for %s: %v", uplinkIFName, ovsErr)
+	}
+	klog.InfoS("Added uplink port on OVS", "ofport", uplinkOFPort)
+	// ExternalEntity is not processed yet, so an empty name is set for entityName in OVSDB,
+	// which will be updated by ExternalNode controller.
+	attachInfo := externalnode.GetOVSAttachInfo(uplinkIFName, uplinkUUID, "", i.externalNodeNamespace, []string{""})
+	// Create host port on OVS.
+	hostIfUUID, ovsErr := i.ovsBridgeClient.CreateInternalPort(hostIFName, 0, "", attachInfo)
+	if ovsErr != nil {
+		return fmt.Errorf("failed to create host port on OVS for %s: %v", hostIFName, ovsErr)
+	}
+
+	// Manual clean up of OVS configurations is required, when agent exits abruptly.
+	defer func() {
+		if !success {
+			klog.InfoS("Deleting port on OVS", "hostIfUUID", hostIfUUID)
+			if ovsErr := i.ovsBridgeClient.DeletePort(hostIfUUID); ovsErr != nil {
+				klog.ErrorS(ovsErr, "Failed to delete port on OVS", "hostIfUUID", hostIfUUID)
+			}
+		}
+	}()
+
+	// Query the host port to check if its created
+	hostOFPort, ovsErr := i.ovsBridgeClient.GetOFPort(hostIFName, false)
+	if ovsErr != nil {
+		return fmt.Errorf("failed to get ofport for host interface %s: %v", hostIFName, ovsErr)
+	}
+	klog.InfoS("Added host port on OVS", "ofport", hostOFPort)
+	success = true
+	return nil
```

Comment:
Shouldn't it call `i.setOVSDatapath()`? I think previous code did it.

Reply:
By mistake I added this in another function. 
Good catch. Updated it.